### ISSUE TITLE
Statistics out of date

### ISSUE THREAD

**Issue by patchthecode** [CocoaPods/cocoapods-stats#32]
A couple things i noticed about the cocoapod site. 

1. It looks like the stats are not up to date? The download count of some cocoapods has been stuck at that number for a while

2. I not sure if the search has become broken?
If i do a search for a pod, it looks like it is displaying old versions of pods. It looks like it is no longer searching by Quality number.

Any help, we use the stats and search feature to increase adoption.

**Comment by pavankataria** [comment]
I too am experiencing the same issue. The download count is definitely stuck on some of the cocoapods, including mine. https://cocoapods.org/pods/SwiftDataTables

I tried contacting some of the guys on twitter but no response yet.

**Comment by segiddins** [comment]
cc @DanToml

**Comment by DanToml** [comment]
Sorry about that - We're running out of storage space in the DB cluster again, and I'm trying to avoid exponentially growing costs (as I am currently paying out of pocket for that cluster).

I'm in the process of figuring out a better way to aggregate the data, to avoid having to shut down the feature.

**Comment by pavankataria** [comment]
I appreciate it costs you money. Out of curiosity does this mean all stats data collected from the past couple of months are lost? What are your costs, maybe we can set up some sort of patreon? I'm sure there'd be folks including myself who would donate for the cause.

**Comment by pavankataria** [comment]
?

**Comment by lucdion** [comment]
@DanToml, this feature is really awesome for open source developers and users. It gives a good idea about the traction of a project. I would also donate to keep this feature alive.

**Comment by pavankataria** [comment]
Maybe an open collective can be created for this purpose and transparency of the cost?

**Comment by pavankataria** [comment]
Is there any update on this?

**Comment by orta** [comment]
@pavankataria please slow down your requests, people are not saying things they can't commit to.

If you want to improve stats, come and help - it has always been OSS and available for people who want to work on it, otherwise it'll happen when it happens.

**Comment by lucdion** [comment]
@DanToml @orta, what if only the total number of downloads is kept? It wouldn't require a lot of spaces, and it would enough. Pods owner could if they want generates the other stats on their repo.
 I offer my help.

**Comment by DanToml** [comment]
@lucidion we could potentially have that, but imo the interesting data is the unique number of app targets, because people pod install from clean trees a lot, which makes raw downloads less interesting.

**Comment by lucdion** [comment]
I agree with @DanToml, but I found this as a temporary solution until the whole system is back on track. At least we would have the total number of downloads, which doesn't require to keep a lot of data, which seems to be the issue with the current solution.

From a developer point of view, it is better to have at least the total number of downloads then nothing.

Also, the number of app targets is really great, but it doesn't tell how many active apps are really using a pod. 

I'm just proposing a short solution that doesn't require to keep a lot of data but still providing a measure of the usage of a pod.

**Comment by pavankataria** [comment]
@DanToml so is my suggestion of providing financial support out of the question by way of an open collective?

**Comment by DanToml** [comment]
@pavankataria It's not completely out of the question, but it has some fairly complicated tax implications that I would need to understand first.

Currently, it's more likely that we'll be shutting down CocoaPods Stats unless we can get support from a cloud provider.

**Comment by lucdion** [comment]
That would be really sad to loose CocoaPods Stats 😕

**Comment by Zyphrax** [comment]
CocoaPods is pretty well known, if someone has some contacts at Amazon or Azure, I wouldn't be surprised that they are very willing to sponsor storage / instances for this. Maybe for a small logo in a bottom corner?

This is great marketing for them, targeted at exactly the audience they are interested in.

**Comment by dcramer** [comment]
Hey folks did we effectively lose stats due to cost? Maybe we can help if thats the case, as they're definitely valuable to companies like us. Feel free to contact me directly if I can help. my first name @sentry.io.

**Comment by pavankataria** [comment]
@dcramer very generous of you. Are there talks in progress in regards to this?

**Comment by bryankeller** [comment]
+1, what happened to stats? Could we at least get some official word on what happened to this feature? It's pretty important to have this data as a library maintainer. When did it go offline / is there any way to get data for 2019?

**Comment by orta** [comment]
Both handling money and on-going maintenance were getting stressful for stats, so I eventually removed the UI from the front-end back in March with https://github.com/CocoaPods/cocoapods.org/pull/411 after we'd not had a working db for the stats in a year or two.

There just aren't folks with free time to maintain any non-critical parts of CocoaPods web infra I'm afraid, especially stats which required hosting our own non-trivial redshift db.
